synfig-studio/src/gui/dialogs/dialog_pluginmanager.cpp

```diff
@@ -0,0 +1,290 @@
+
+        plugin_name->set_text(plugin.name.get());
+
+        plugin_list_item->pack_start(*plugin_name, Gtk::PACK_SHRINK, 20);
+        plugin_list_item->pack_end(*plugin_option_box, Gtk::PACK_SHRINK, 20);
+        
+        plugin_list_box.add(*plugin_list_item);
+
+    }
+    plugin_list_box.show_all_children();
+}
+
+std::string extract_plugin_name(std::istream &fileStream)
+{
+    xmlpp::DomParser parser;
+    parser.set_substitute_entities();
+    parser.parse_stream(fileStream);
+
+    const xmlpp::Node* pNode = parser.get_document()->get_root_node();
+    if (std::string(pNode->get_name()) != std::string("plugin") ) {
+        synfig::warning("Invalid plugin.xml file (missing root <plugin>)");
+        return std::string("");
+    }
+
+    auto execlist = pNode->find("./exec");
+    if (!execlist.empty())
+    {
+        for (const xmlpp::Node* node : pNode->find("./name") ) {
+            const xmlpp::Element* element = dynamic_cast<const xmlpp::Element*>(node);
+            std::string lang = element->get_attribute_value("lang");
+            if (lang.empty()) {
+                return element->get_child_text()->get_content();
+            }
+        }
+    }
+}
+void
+Dialog_PluginManager::on_install_plugin_button_clicked()
+{
+    // Show the dialog and wait for a user response
+    int result = plugin_file_dialog.run();
+    switch (result) {
+        case (Gtk::RESPONSE_OK):
+        { 
+            std::vector<std::string> files;
+            PluginZipStatus pluginStatus = PluginZipStatus::PLUGIN_ZIP_NO_PLUGIN_XML;
+            etl::handle<synfig::FileSystemNative> native_fs = FileSystemNative::instance();
+            etl::handle<synfig::FileContainerZip> zip_fs = new FileContainerZip();
+            std::string zip_filename = plugin_file_dialog.get_filename();
+            filesystem::Path path_to_user_plugins = synfigapp::Main::get_user_app_directory() / filesystem::Path("plugins");
+            std::string output_path ;
+            std::string plugin_metadata_file;
+            std::string plugin_name ;
+
+            if (!zip_fs->open(plugin_file_dialog.get_filename())) {
+                synfig::error("Failed to open zip file: " + zip_filename);
+                return;
+            }
+            zip_fs->directory_scan("", files);
+            for (const auto& file : files) {
+                if (file == "plugin.xml") {
+                    plugin_metadata_file = file;
+                    break;
+                } else if(zip_fs->is_directory(file)) {
+                    std::vector<std::string> child_files;
+                    if (zip_fs->directory_scan(file, child_files)) {
+                        for (auto& innerFile : child_files) {
+                            if (innerFile == "plugin.xml") {
+                                plugin_metadata_file = file + "/" + innerFile;
+                                break;
+                            }
+                        }
```

Comment:
I don't know if `std::find(child_files.begin(), child_files.end(), "plugin.xml")` would be better.
What do you think?